cmd/river/go.mod

```diff
@@ -1,16 +1,20 @@
 module github.com/riverqueue/river/cmd/river
 
-go 1.22.5
+go 1.21
```

Comment:
@bgentry I changed all of these to 1.21 instead of a specific patch version like 1.21.13.

1.21 == 1.21.0

I think this is probably right because it gives maximum flexibility for people installing the package. They should still use the latest possible patch version of course, but it's not always possible due to their local build toolchain. Specifying 1.21.0 just seems like the much easier choice rather than trying to find a minimum patch version that's most universally available.

Do you agree?